table/sst_file_writer.cc

```diff
@@ -368,6 +371,56 @@ Status SstFileWriter::Finish(ExternalSstFileInfo* file_info) {
     r->file_info.file_checksum_func_name =
         r->file_writer->GetFileChecksumFuncName();
   }
+
+  if (s.ok() && r->mutable_cf_options.paranoid_file_checks) {
+    std::unique_ptr<TableReader> table_reader;
+    std::unique_ptr<FSRandomAccessFile> sst_file;
+    std::unique_ptr<RandomAccessFileReader> sst_file_reader;
+
+    s = r->ioptions.fs->NewRandomAccessFile(r->file_info.file_path,
+                                            r->env_options, &sst_file, nullptr);
+    if (!s.ok()) {
+      return s;
+    }
+
+    sst_file_reader.reset(new RandomAccessFileReader(
+        std::move(sst_file), r->file_info.file_path, nullptr, nullptr));
```

Comment:
Probably should add comment for argument names for the two `nullptr`

Reply:
Addressed in subsequent commit